### PR TITLE
Update go to 16 and bump dependencies

### PR DESCRIPTION
**Pull request by fabritsius**

### CODE REVIEW

**Review by leo-ri — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -27,7 +27,6 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4-205.1626828526
 RUN microdnf install yum &&\
     yum -y update-minimal --security --sec-severity=Important --sec-severity=Critical &&\
     yum clean all &&\
-    microdnf remove yum &&\
```
[the remark this answers is not in the document]
Comment (fabritsius) [reply Dockerfile:30]: It seems `yum` became protected in `go1.16` image:
```
Problem: The operation would result in removing the following protected packages: yum
```
It doesn't allow to remove `yum` maybe there is an alternative walk-around.

File: .github/actions/gen-install-scripts/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.15
+FROM golang:1.16
 
 ENV KUBECTL_VERSION 1.18.12
 
-RUN go get github.com/mikefarah/yq
+RUN GO111MODULE=on go get github.com/mikefarah/yq/v4
```
[the remark this answers is not in the document]
Comment (fabritsius) [reply .github/actions/gen-install-scripts/Dockerfile:5]: Not sure. We should if all other installations also use modules. I haven't tried, this is just a safer initial change.

[the remark this answers is not in the document]
Comment (fabritsius) [reply .github/actions/gen-install-scripts/Dockerfile:5]: No harm in trying. Thanks you)

[the remark this answers is not in the document]
Comment (fabritsius) [reply .github/actions/gen-install-scripts/Dockerfile]: Works 👍🏻

**Review by leo-ri — APPROVED** [review]

**Pull request merged**